### DISCUSSION TITLE
how to integrate library with django

### DISCUSSION THREAD

**Discussion by SebaRGFSL** [sns-sdks/python-facebook#207] [Q&A] [answered]
i'm trying to integrate the library with django.
So far I managed to successfully go through the entire login process in one view, but I'm struggling to get the user access token with the `api.exchange_user_access_token()` method. As far as I understand the process is the following:

- Generate API instance with app-id and app-secret
- call `api.get_authorization_url()` (this is another django view)

What I don't fully understand is where and with which parameter I should call `api.exchange_user_access_token()` if it's after the redirection or in the same first view where I'm instatiating the class for the first time

**Comment by MerleLiuKun** [comment, marked as answer]
Hi,

`api.exchange_user_access_token()` need called when `facebook` redirect to `redirect_uri` domain.

Then you will exchange user acess token with callback parameters.
